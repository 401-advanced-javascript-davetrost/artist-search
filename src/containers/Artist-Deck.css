.ArtistDeck {
  display: flex;
  flex-direction: row;
  justify-content: center;

  button {
    width: 15%;
    height: 30px;
    margin-top: 15%;
    background: #e97e31;
    color: whitesmoke;
    font-size: 1.2em;
    text-transform: uppercase;
    font-family: 'Questrial', sans-serif;
    font-weight: 700;
    &:hover {
      cursor: pointer;
      background-color: rgba(233, 126, 49, 0.75);
    }
  }

  ul {
    width: 50em;
    text-align: center;
    font-family: 'Questrial', sans-serif;
    line-height: 1.1em;
  }

  h2:hover {
    color: #459DB4;
    cursor: pointer;
  }
}