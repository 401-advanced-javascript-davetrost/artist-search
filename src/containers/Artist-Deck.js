import React, { memo } from 'react';
import PropTypes from 'prop-types';
import ArtistCard from '../components/Artist-Card';
import styles from './Artist-Deck.css';

function ArtistDeck({ artists, handleNext, handleBack }) {
  const artistCards = artists.map(artist => {
    return (
      <li key={artist.id}>
        <ArtistCard id={artist.id} artist={artist.name} />
      </li>
    );
  });

  return (
    <section className={styles.ArtistDeck}>
      <button name="back" onClick={handleBack}>Back</button>
      <ul>
        {artistCards}
      </ul>
      <button name="next" onClick={handleNext}>Next</button>
    </section>
  );
}

ArtistDeck.propTypes = {
  artists: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired
    })).isRequired,
  handleBack: PropTypes.func.isRequired,
  handleNext: PropTypes.func.isRequired
};

export const MemoArtistDeck = memo(ArtistDeck);
