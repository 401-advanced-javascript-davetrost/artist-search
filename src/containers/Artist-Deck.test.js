import React from 'react';
import { shallow } from 'enzyme';
import { MemoArtistDeck } from './Artist-Deck';

describe('Artist Deck component', () => {

  it('renders a list of artists', () => {
    const backFunc = jest.fn();
    const nextFunc = jest.fn();  
    const wrapper = shallow(<MemoArtistDeck 
      artists={[
        { id: '1', name: 'Hanky' },
        { id: '2', name: 'Panky' },
      ]}
      handleBack={backFunc}
      handleNext={nextFunc}
    />);
    expect(wrapper).toMatchSnapshot();

    wrapper.find('button[name="back"]').simulate('click');
    expect(backFunc).toHaveBeenCalledTimes(1);
    wrapper.find('button[name="next"]').simulate('click');
    expect(nextFunc).toHaveBeenCalledTimes(1);
  });
});
