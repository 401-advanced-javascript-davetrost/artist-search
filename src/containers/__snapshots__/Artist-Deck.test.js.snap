// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist Deck component renders a list of artists 1`] = `
<section
  className="ArtistDeck"
>
  <button
    name="back"
    onClick={[MockFunction]}
  >
    Back
  </button>
  <ul>
    <li
      key="1"
    >
      <ArtistCard
        artist="Hanky"
        id="1"
      />
    </li>
    <li
      key="2"
    >
      <ArtistCard
        artist="Panky"
        id="2"
      />
    </li>
  </ul>
  <button
    name="next"
    onClick={[MockFunction]}
  >
    Next
  </button>
</section>
`;
